dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/template/components/ComponentFactoryContainer.kt

```diff
@@ -0,0 +1,10 @@
+package org.dataland.frameworktoolbox.template.components
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+/**
+ * A surrogate component for retrieving an autowired ordered list of TemplateComponentFactories
+ */
+@Component
+class ComponentFactoryContainer(@Autowired val factories: List<TemplateComponentFactory>)
```

Comment:
Why do we use this wrapper? If we can autowire the list of `TemplateComponentFactory`s, why does the bean exist in the first place?

Reply:
@MarcTM01

Reply:
This wrapper is required to ensure that the list of factories is ordered according to the ordering annotations on the spring beans.

Reply:
Ok. On https://www.baeldung.com/spring-order they do not proceed like that. But as it is working fine now, resolved.